### PR TITLE
Add Dog and DogTest

### PR DESCRIPTION
**Pull request by uthmanmoh**

### CODE REVIEW

File: src/main/java/Dog.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    }
+
+    @Override
+    public String sound() {
+        return "Bark";
+    }
+
+    @Override
+    public int getPrice() {
+        return 1;
```
Comment (abdullah-sha256) [inline src/main/java/Dog.java:10]: are dogs really that cheap?

File: src/main/java/Dog.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+public class Dog implements Domesticatable, Tradable {
+    public Dog() {
+
```
Comment (abdullah-sha256) [inline src/main/java/Dog.java]: you dont need this

**Review by nxaden — CHANGES_REQUESTED** [review]
fix it i dare u

File: src/main/java/Dog.java (hunk 2 above)
Comment (uthmanmoh) [reply src/main/java/Dog.java]: good point!

File: src/main/java/Dog.java (hunk 1 above)
Comment (uthmanmoh) [reply src/main/java/Dog.java:10]: yes

File: src/main/java/Dog.java (hunk 2 above)
Comment (sourenrex) [reply src/main/java/Dog.java]: True!

File: src/main/java/Dog.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    }
+
+    @Override
+    public int getPrice() {
+        return 1;
+    }
+
+    @Override
+    public String toString(){
+        return "I am a Dog";
```
Comment (abdullah-sha256) [inline src/main/java/Dog.java:15]: why do u have this?

**Review by nxaden — APPROVED** [review]
Approved

**Pull request merged**